app/decorators/service_provider_session_decorator.rb

```diff
@@ -33,6 +33,11 @@ def nav_partial
     'shared/nav_branded'
   end
 
+  def forgot_password_partial
+    'reset_password/link' if loa3?
+    'forgot_password/link'
```

Comment:
this will always return `forgot_password/link` since there's no `return`

I think this would be clearer in the block form:

```
if loa3?
  'reset_password/link'
else
  'forgot_password/link'
end
```